CMakeLists.txt

```diff
@@ -41,6 +41,9 @@ endif ()
 add_library(netcdf)
 add_library(netCDF::netcdf ALIAS netcdf)
 
+set_target_properties(netcdf PROPERTIES
+    POSITION_INDEPENDENT_CODE ON
```

Comment:
`pic` is added by default on shared libraries. Shared vs static is controlled via `BUILD_SHARED_LIBS`.

Also it can be helpful to set `cmake_minimum_required(VERSION 3.13...3.30)` the latter part sets policy levels when you have any `if(CMAKE_VERSION)`. I think there is one policy of interest around the library targets, but not sure what version it involves. Try out with latest cmake version requirement and work your way backwards after that.